src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -222,8 +225,13 @@ def construct_payload(self):
 
             # Check if updating existing container
             updating_existing_container = False
+            if self.get_argument_force_single_container_updates():
+                # Remove n-1 containers where n is the number of containers in the containerapp and replace remaining container with the new container
+                # Fails if all containers are removed
+                while (len(self.new_containerapp["properties"]["template"]["containers"]) > 1):
+                    self.new_containerapp["properties"]["template"]["containers"].pop()
```

Comment:
When will the value of `force_single_container_updates` be `True`?

I check the previous logic, currently only when `No container registry provided. Let's use the default container registry through Cloud Build.`, the value can be `True`.
Why you pop all other containers? Will it cause existing containers delete that customer not aware to delete? @snehapar9  @daniv-msft 
```
        if self.should_create_acr:
            # No container registry provided. Let's use the default container registry through Cloud Build.
            self.image = self.build_container_from_source_with_cloud_build_service(source, build_env_vars, location)
            return True
```

Reply:
@Greedygre the goal here is to delete all existing containers for cloud build scenarios to ensure the containerapp only has one container with the newly built image. 
The `force_single_container_updates` argument is set only for cloud build scenarios.

Reply:
@snehapar9 

`set only for cloud build scenario` is here?

```
    if self.should_create_acr:
            # No container registry provided. Let's use the default container registry through Cloud Build.
            self.image = self.build_container_from_source_with_cloud_build_service(source, build_env_vars, location)
            return True
```

Is there any warning or info in CLI for customer or ant Doc that about `delete all existing containers for cloud build scenarios to ensure the containerapp only has one container with the newly built image.`?

Reply:
Customers would not have control over it because they are using the cloud build scenario, so we want to make sure there is only the container with the image built by the source to cloud builder.
Not sure if it necessary for the customer to be warned? @daniv-msft any thoughts on this?

Reply:
Here the ARM update to the container app will create a new revision, so if the customers encounter any issues they can always go back to their previous version. That said here in this command the customers are asking us to redeploy on their Container App with the source code provided. I don't think that many of them would be surprised to lose their current containers (and if they did they could always roll back to the previous revision).